### PR TITLE
[NOD-1119] Refactor main, and remove p2p layer from it

### PR DESCRIPTION
**Pull request by svarogg**

### CODE REVIEW

**Comment by svarogg** [comment]
Rebuild after update in master

File: blockdag/common_test.go [hunk 1]
```diff
@@ -98,8 +98,8 @@ func newTestDAG(params *dagconfig.Params) *BlockDAG {
 	index := newBlockIndex(params)
 	targetTimePerBlock := int64(params.TargetTimePerBlock / time.Second)
 	dag := &BlockDAG{
-		dagParams:                      params,
-		timeSource:                     NewTimeSource(),
+		Params:                         params,
+		TimeSource:                     NewTimeSource(),
```
Comment (someone235) [inline blockdag/common_test.go:102]: Instead of exporting you can use dag.Now

File: server/server.go [hunk 2]
```diff
@@ -68,36 +70,85 @@ func (s *Server) Stop() error {
...
+	var acceptanceIndex *indexers.AcceptanceIndex
+	if config.ActiveConfig().AcceptanceIndex {
+		log.Info("acceptance index is enabled")
+		indexes = append(indexes, acceptanceIndex)
 	}
-	blockTemplateGenerator := mining.NewBlkTmplGenerator(&policy,
-		s.p2pServer.DAGParams, s.p2pServer.TxMemPool, s.p2pServer.DAG, s.p2pServer.TimeSource, s.p2pServer.SigCache)
+	// Create an index manager if any of the optional indexes are enabled.
+	var indexManager blockdag.IndexManager
+	if len(indexes) > 0 {
```
Comment (someone235) [inline server/server.go:115]: You can flip the condition and `return nil, nil`

File: server/server.go [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
 	"github.com/kaspanet/kaspad/config"
 	"github.com/kaspanet/kaspad/dagconfig"
 	"github.com/kaspanet/kaspad/mempool"
 	"github.com/kaspanet/kaspad/mining"
-	"github.com/kaspanet/kaspad/server/p2p"
 	"github.com/kaspanet/kaspad/server/rpc"
 	"github.com/kaspanet/kaspad/signal"
 )
 
 // Server is a wrapper for p2p server and rpc server
```
Comment (stasatdaglabs) [inline server/server.go:20]: This object is now a little bit more than just those two.

File: server/server.go [hunk 4]
```diff
@@ -1,22 +1,28 @@
...
 type Server struct {
```
Comment (stasatdaglabs) [inline server/server.go:21]: This object is not a server. It encompasses Kaspad's entire state. Please consider renaming it.

Comment (stasatdaglabs) [reply server/server.go:21]: We decided that we don't want this object at all. Instead, DAG, mempool, p2pServer, rpcServer, etc. should all be started in Main.

File: kaspad.go [hunk 5]
```diff
@@ -1,291 +1,159 @@
...
-	interrupt := signal.InterruptListener()
+	started, shutdown int32
+}
 
-	// Load configuration and parse command line. This function also
-	// initializes logging and configures it accordingly.
-	err := config.LoadAndSetActiveConfig()
-	if err != nil {
-		return err
+// Start begins accepting connections from peers.
```
Comment (someone235) [inline kaspad.go:25]: Weird comment

File: kaspad.go [hunk 6]
```diff
@@ -1,291 +1,158 @@
...
 )
 
-var (
-	cfg *config.Config
-)
-
-// winServiceMain is only invoked on Windows. It detects when kaspad is running
-// as a service and reacts accordingly.
-var winServiceMain func() (bool, error)
+// Kaspad is a wrapper for all the kaspad services
```
Comment (stasatdaglabs) [inline kaspad.go:18]: Please write kaspad in lowercase here.

File: main.go [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+
+	if cfg.ResetDatabase {
+		err := removeDatabase()
+		if err != nil {
+			log.Errorf("%s", err)
+			return err
+		}
+	}
+
+	// Open the database
```
Comment (stasatdaglabs) [inline main.go:104]: Please consider opening the database in kaspad's start.

**Pull request merged**